src/game/phase/phase_pause.c

```diff
@@ -152,17 +105,83 @@ bool Game_Pause(void)
 
     Music_Pause();
     Sound_PauseAll();
-    const GAME_STATUS old_status = Game_GetStatus();
+
+    m_OldGameStatus = Game_GetStatus();
     Game_SetStatus(GS_IN_PAUSE);
 
     Output_FadeToSemiBlack(true);
-    int32_t select = Game_Pause_Loop();
+
+    m_State = 0;
+}
+
+static void Phase_Pause_End(void)
+{
     Output_FadeToTransparent(true);
 
     Music_Unpause();
     Sound_UnpauseAll();
     Requester_Remove(&m_PauseRequester);
-    Game_Pause_RemoveText();
-    Game_SetStatus(old_status);
-    return select < 0;
+    Phase_Pause_RemoveText();
 }
+
+static int32_t Phase_Pause_Control(int32_t nframes)
+{
+    Phase_Pause_UpdateText();
+
+    Input_Update();
+    Shell_ProcessInput();
+    Game_ProcessInput();
+
+    switch (m_State) {
+    case 0:
```

Comment:
Should we use enums for the case statement's pause states?

Reply:
Added